src/app/modules/group/components/add-sub-group/add-sub-group.component.ts

```diff
@@ -0,0 +1,82 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Subject, timer } from 'rxjs';
+import { debounce, map, switchMap, tap } from 'rxjs/operators';
+import { Group } from 'src/app/core/components/group-nav-tree/group';
+import { AddedContent, NewContentType } from 'src/app/modules/shared-components/components/add-content/add-content.component';
+import { SearchGroupService } from '../../http-services/search-group.service';
+
+type GroupType = 'Class'|'Team'|'Club'|'Friends'|'Other';
+
+@Component({
+  selector: 'alg-add-sub-group',
+  templateUrl: './add-sub-group.component.html',
+  styleUrls: [ './add-sub-group.component.scss' ]
+})
+export class AddSubGroupComponent {
+
+  @Input() group?: Group;
+
+  @Output() addGroup = new EventEmitter<AddedContent<GroupType>>();
+
+  groupsFound: {
+    type: GroupType,
+    title: string,
+    description: string|null,
+  }[] = [];
+
+  state: 'loading' | 'ready' = 'loading';
+
+  private dataFetching = new Subject<string>();
+
+  newGroupTypes: NewContentType<GroupType>[] = [
+    {
+      type: 'Class',
+      icon: 'fa fa-book',
+      title: $localize`Class`,
+      description: $localize`Class`,
+    },
+    {
+      type: 'Club',
+      icon: 'fa fa-book',
+      title: $localize`Club`,
+      description: $localize`Club`,
+    },
+    {
+      type: 'Friends',
+      icon: 'fa fa-users',
+      title: $localize`Friends`,
+      description: $localize`Friends`,
+    },
+    {
+      type: 'Other',
+      icon: 'fa fa-book',
+      title: $localize`Other`,
+      description: $localize`Other`,
+    },
+  ];
```

Comment:
I'm not sure what description and icon to have for each of the new group category